packages/graph/hash_graph/lib/graph/src/api/rest/middleware.rs

```diff
@@ -129,7 +148,7 @@ fn parse_x_forwarded_for(headers: &http::HeaderMap) -> Option<Cow<'_, str>> {
 }
 
 // Based on https://github.com/tokio-rs/axum/pull/769
-pub fn span_maker(request: &hyper::Request<hyper::Body>) -> tracing::Span {
+fn span_maker(request: &hyper::Request<hyper::Body>) -> tracing::Span {
     let target = request.uri();
     let scheme: Cow<'static, str> = target.scheme().map_or_else(
         || "HTTP".into(),
```

Comment:
Also slightly too far from the diff, why do we need the `"http".into()` and `"https".into()`? Can't we just use `to_string` on all of them? Or is that to save us a few allocations with the `Cow` or something?

Reply:
Since we're returning a cow, we wouldn't allocate "HTTP" or "HTTPS" as a String using `into` here